contracts/schemes/TokenTrade.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity ^0.6.10;
+// SPDX-License-Identifier: GPL-3.0
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract TokenTrade is VotingMachineCallbacks, ProposalExecuteInterface {
+    using SafeMath for uint;
+    using SafeERC20 for IERC20;
+
+    event TokenTradeProposed(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address indexed _beneficiary,
+        IERC20 _sendToken,
+        uint256 _sendTokenAmount,
+        IERC20 _receiveToken,
+        uint256 _receiveTokenAmount
+    );
+
+    event TokenTradeProposalExecuted(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        IERC20 _sendToken,
+        uint256 _sendTokenAmount,
+        IERC20 _receiveToken,
+        uint256 _receiveTokenAmount
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        address beneficiary;
+        IERC20 sendToken;
+        uint256 sendTokenAmount;
+        IERC20 receiveToken;
+        uint256 receiveTokenAmount;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _votingParams genesisProtocol parameters - valid only if _voteParamsHash is zero
+     * @param _voteOnBehalf genesisProtocol parameter - valid only if _voteParamsHash is zero
+     * @param _voteParamsHash voting machine parameters.
+     */
+    function initialize(
+        Avatar _avatar,
```

Comment:
I think it should get also the tradingToken

Reply:
Why not allow any token? What's the reason to limit it?

Reply:
1. easy in the UI.
2. simple code
3. more secure